src/util/imagefiledata.cpp

```diff
@@ -0,0 +1,65 @@
+#include "util/imagefiledata.h"
+
+#include <QBuffer>
+#include <QByteArray>
+#include <QFile>
+#include <QImage>
+#include <QImageReader>
+
+#include "util/assert.h"
+
+ImageFileData::ImageFileData(
+        const QByteArray& coverArtBytes, const QByteArray& coverArtFormat)
+        : QImage(QImage::fromData(coverArtBytes)),
+          m_coverArtBytes(coverArtBytes) {
+    m_coverArtFormat = coverArtFormat.isNull() ? getFormat(m_coverArtBytes) : coverArtFormat;
+}
+
+//static
+ImageFileData ImageFileData::fromFilePath(const QString& coverArtFilePath) {
+    QFile coverArtFile = QFile(coverArtFilePath);
+    coverArtFile.open(QIODevice::ReadOnly);
+    QByteArray coverArtBytes = coverArtFile.readAll();
+    coverArtFile.close();
+    QByteArray coverArtFormat = getFormat(coverArtBytes);
+    return ImageFileData(coverArtBytes, coverArtFormat);
+}
+
+//static
+QByteArray ImageFileData::getFormat(const QByteArray& coverArtBytes) {
+    QByteArray bytes = coverArtBytes;
+    QBuffer b = QBuffer(&bytes);
+    b.open(QIODevice::ReadOnly);
+    auto reader = QImageReader(&b);
+    QByteArray coverArtFormat = reader.format();
+    VERIFY_OR_DEBUG_ASSERT(!coverArtFormat.isEmpty()) {
+        return QByteArray{};
+    }
+    return coverArtFormat;
+}
+
+bool ImageFileData::saveFile(const QString& coverArtAbsoluteFilePath) const {
+    QFile coverArtFile(coverArtAbsoluteFilePath);
```

Comment:
this one can default to QImage::save in case m_coverArtBytes is empty.

Reply:
I guess that is done with the last commit